homeassistant/components/torque/config_flow.py

```diff
@@ -0,0 +1,86 @@
+"""Config flow for Torque integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_EMAIL, CONF_NAME
+from homeassistant.core import DOMAIN as HOMEASSISTANT_DOMAIN
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+
+from .const import DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+ISSUE_PLACEHOLDER = {"url": "/config/integrations/dashboard/add?domain=torque"}
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+class TorqueConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Torque."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self._async_abort_entries_match(
+                {
+                    CONF_NAME: user_input[CONF_NAME],
+                    CONF_EMAIL: user_input[CONF_EMAIL],
+                }
+            )
+
+            await self.async_set_unique_id(user_input[CONF_NAME])
+            self._abort_if_unique_id_configured()
```

Comment:
Oh and we abort if the unique_id is already used, which isn't since there can only be one. Name is a bad unique_id btw

Reply:
Yeah, I wasn't entirely sure what to put as the unique id. 🤔 Or whether to just not set one?

Reply:
Unique_id should be set to something unique that can't change. Mac address, serial number, customer id, something along those lines.

Bad examples are names, emailaddresses (if they can change, otherwise they are good), IP addresses

Reply:
I think email should be fine, as it can't be changed (without removing the integration)? 🤔 Or is that still an issue?

Reply:
What do you use it for in the first place? How does Torque communicate? from what I understand now you have a webhook, but how does that work

Reply:
The "email" is just an email (or any string), that is used for validating the query data sent to the endpoint. It's just a requirement in the Torque app. We don't seem to really use it, other than checking the email matches the one in the query.

I guess in a way, it's like an auth token (in addition to the Bearer token for HA's API). So it's probably best actually to not use it as a unique id.

Reply:
Another value that can be used is the ID torque generates. (It'll be the query parameter 'id')